client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/test/payment-method-add.jsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, act } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { Provider } from 'react-redux';
+import { createStore } from 'redux';
+import PaymentMethodAdd from '../index';
+
+describe( '<PaymentMethodAdd>', () => {
+	test( 'should render correctly and match the snapshot', async () => {
+		const promise = Promise.resolve();
+		const initialState = {
+			ui: { section: 'test' },
+			documentHead: { unreadCount: 1 },
+		};
+
+		const store = createStore( ( state ) => state, initialState );
+
+		const { container, getByText } = render(
+			<Provider store={ store }>
+				<PaymentMethodAdd />
+			</Provider>
+		);
+
+		const [ aTag ] = container.getElementsByClassName( 'payment-method-add__back-button' );
+		const href = 'https://example.com/partner-portal/payment-methods/';
+
+		expect( aTag ).toHaveProperty( 'href', href );
+
+		expect( getByText( 'Credit card details' ) ).toBeInTheDocument();
+
+		await act( () => promise );
```

Comment:
Why is this needed here?

Reply:
The component `PaymentMethodAdd` has a `StripeSetupIntentIdProvider` which gets updated after the component is mounted. When testing, code that causes React state updates should be wrapped into an `[act](https://reactjs.org/docs/test-utils.html#act)`. 

We get errors like this otherwise:

<img width="1046" alt="Screenshot 2022-04-21 at 11 47 02 AM" src="https://user-images.githubusercontent.com/10586875/164386753-51c60b0e-82d8-4a87-a00b-09eb8df629a8.png">

This line is basically added so that it waits for all the promises to be resolved. Without this, the tests pass, but we get this Warning.

Reply:
Got it. Thank you.